src/serializer/serializer.js

```diff
@@ -119,12 +120,45 @@ export class Serializer {
     if (timingStats !== undefined) timingStats.deepTraversalTime = Date.now();
     let residualHeapVisitor = new ResidualHeapVisitor(this.realm, this.logger, this.modules);
     residualHeapVisitor.visitRoots();
+    this.realm.wrapInGlobalEnv(() => this.realm.evaluateForEffects(() => {
```

Comment:
Can we avoid evaluating the functions again? `checkThatFunctionsAreIndependent` should have done that already, we should just need to keep the effects around.